src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1127,7 +1088,7 @@ overlay_packages (RpmOstreeSysrootUpgrader *self,
    * that the user wants to stop overlaying them), so we have another
    * optimization here for that case.
    */
-  if (g_strv_length (self->requested_packages) == 0)
+  if (!rpmostree_origin_is_locally_assembled (self->origin))
```

Comment:
This is one spot where we don't want to sub out I think. If the initramfs is local, but there are no longer packages to layer, we miss out on this optimization.